public/app/plugins/datasource/prometheus/legend.test.ts

```diff
@@ -20,11 +20,25 @@ describe('renderLegendFormat()', () => {
     expect(renderLegendFormat('{{ a }}', labels)).toEqual('AAA');
   });
 
+  it('Missing label', () => {
+    // it's unclear if this is expected behavior
+    expect(renderLegendFormat('value: {{c}}', labels)).toEqual('value: c');
+    expect(renderLegendFormat('{{        }}', labels)).toEqual('');
+  });
+
+  it('Nested brackets', () => {
+    // it's unclear if this is expected behavior
+    expect(renderLegendFormat('{{{a}}}', labels)).toEqual('{AAA}');
+    expect(renderLegendFormat('{{{{a}}}}', labels)).toEqual('{{AAA}}');
+    expect(renderLegendFormat('{{ {{ a }} }}', labels)).toEqual('{{ AAA }}');
+  });
+
   it('Bad syntax', () => {
     expect(renderLegendFormat('value: {{a}', labels)).toEqual('value: {{a}');
     expect(renderLegendFormat('value: {a}}}', labels)).toEqual('value: {a}}}');
-
-    // Current behavior -- not sure if expected or not
-    expect(renderLegendFormat('value: {{{a}}}', labels)).toEqual('value: {a}');
```

Comment:
Feels like we should keep this expect to make sure we have not changed any behavior

Reply:
@mckn after thinking about this more, i am not sure if we can support the current behavior and still have it pass the ReDoS checks since i believe the goal is to limit the nested matching of `{` and `}` 

i'm definitely not a regex expert, so it might need a 👀 from someone with more advanced regex knowledge. i was using https://devina.io/redos-checker to test